indigo/tests/test_refsfinders.py

```diff
@@ -0,0 +1,202 @@
+        self.assertEqual(document.content, expected.content)
+
+    def test_section_notfound(self):
+        document = Document(
+            document_xml=document_fixture(
+                xml="""
+      <section id="section-7">
+        <num>7.</num>
+        <heading>Active ref heading</heading>
+        <content>
+          <p>As given in section 26, which isn't in this document, blah.</p>
+        </content>
+      </section>
+      <section id="section-35">
+        <num>35.</num>
+        <heading>Not the section we want</heading>
+        <content>
+          <p>Not the provision you're looking for.</p>
+        </content>
+      </section>
+        """
+            ),
+            language=self.eng)
+
+        expected_content = document.content
+        self.section_refs_finder.find_references_in_document(document)
+        root = etree.fromstring(expected_content)
+        expected_content = etree.tostring(root, encoding='utf-8').decode('utf-8')
+        self.assertEqual(document.content, expected_content)
+
+    def test_section_external(self):
+        document = Document(
+            document_xml=document_fixture(
+                xml="""
+      <section id="section-7">
+        <num>7.</num>
+        <heading>Active ref heading</heading>
+        <content>
+          <p>As given in section 26 of Act 5 of 2012, blah.</p>
+          <p>As given in section 26 of the Nursing Act, blah.</p>
+        </content>
+      </section>
+      <section id="section-26">
+        <num>26.</num>
+        <heading>Passive ref heading</heading>
+        <content>
+          <p>An important provision, but not the one referred to above.</p>
+        </content>
+      </section>
+        """
+            ),
+            language=self.eng)
+
+        expected_content = document.content
+        self.section_refs_finder.find_references_in_document(document)
+        root = etree.fromstring(expected_content)
+        expected_content = etree.tostring(root, encoding='utf-8').decode('utf-8')
+        self.assertEqual(document.content, expected_content)
+
+    def test_section_external_in_tail(self):
+        document = Document(
+            document_xml=document_fixture(
+                xml="""
+      <section id="section-7">
+        <num>7.</num>
+        <heading>Active ref heading</heading>
+        <content>
+          <p>As <i>given</i> in (we're now in a tail) section 26 of Act 5 of 2012, blah.</p>
+          <p>As <i>given</i> in (we're now in a tail) section 26 of the Nursing Act, blah.</p>
+        </content>
+      </section>
+      <section id="section-26">
+        <num>26.</num>
+        <heading>Passive ref heading</heading>
+        <content>
+          <p>An important provision, but not the one referred to above.</p>
+        </content>
+      </section>
+        """
+            ),
+            language=self.eng)
+
+        expected_content = document.content
```

Comment:
?

Reply:
This test doesn't look like it has the correct expected value?

Reply:
no it does, we're testing here to make sure that external section refs don't get marked up by accident